eng/build-utils.ps1

```diff
@@ -237,16 +237,30 @@ function Make-BootstrapBuild() {
     Create-Directory $dir
 
     # prepare FsLex and Fsyacc and AssemblyCheck
-    Run-MSBuild "$RepoRoot\src\buildtools\buildtools.proj" "/restore /t:Publish /p:PublishWindowsPdb=false" -logFileName "BuildTools" -configuration $bootstrapConfiguration
-    Copy-Item "$ArtifactsDir\bin\fslex\$bootstrapConfiguration\netcoreapp3.0\publish" -Destination "$dir\fslex" -Force -Recurse
-    Copy-Item "$ArtifactsDir\bin\fsyacc\$bootstrapConfiguration\netcoreapp3.0\publish" -Destination "$dir\fsyacc" -Force  -Recurse
-    Copy-Item "$ArtifactsDir\bin\AssemblyCheck\$bootstrapConfiguration\netcoreapp3.0\publish" -Destination "$dir\AssemblyCheck" -Force  -Recurse
+    $dotnetPath = InitializeDotNetCli
+    $dotnetExe = Join-Path $dotnetPath "dotnet.exe"
+    $buildToolsProject = "$RepoRoot\src\buildtools\buildtools.proj"
+
+    $argNoRestore = if ($norestore) { " --no-restore" } else { "" }
+    $argNoIncremental = if ($rebuild) { " --no-incremental" } else { "" }
+
+    $args = "build $buildToolsProject -c $bootstrapConfiguration -v $verbosity -f netcoreapp3.0" + $argNoRestore + $argNoIncremental
+    Exec-Console $dotnetExe $args
+
+    Copy-Item "$ArtifactsDir\bin\fslex\$bootstrapConfiguration\netcoreapp3.0" -Destination "$dir\fslex" -Force -Recurse
+    Copy-Item "$ArtifactsDir\bin\fsyacc\$bootstrapConfiguration\netcoreapp3.0" -Destination "$dir\fsyacc" -Force  -Recurse
+    Copy-Item "$ArtifactsDir\bin\AssemblyCheck\$bootstrapConfiguration\netcoreapp3.0" -Destination "$dir\AssemblyCheck" -Force  -Recurse
 
     # prepare compiler
-    $projectPath = "$RepoRoot\proto.proj"
-    Run-MSBuild $projectPath "/restore /t:Publish /p:TargetFramework=$bootstrapTfm;ProtoTargetFramework=$bootstrapTfm /p:PublishWindowsPdb=false" -logFileName "Bootstrap" -configuration $bootstrapConfiguration
-    Copy-Item "$ArtifactsDir\bin\fsc\$bootstrapConfiguration\$bootstrapTfm\publish" -Destination "$dir\fsc" -Force -Recurse
-    Copy-Item "$ArtifactsDir\bin\fsi\$bootstrapConfiguration\$bootstrapTfm\publish" -Destination "$dir\fsi" -Force -Recurse
+    $protoProject = "$RepoRoot\proto.proj"
+    $args = "build $protoProject -c $bootstrapConfiguration -v $verbosity -f $bootstrapTfm" + $argNoRestore + $argNoIncremental
```

Comment:
Same comment as above for `-f $bootstrapTfm`; `proto.proj` handles passing that through to the projects, so it shouldn't be necessary here.

Reply:
@brettfo,

because we copy from specific directories, I believe we should force the TFM here,   If you look at the lines below this you will see we copy from netcoreapp3.0.  So yes, I could rely on what is in the project, but I think this is a bit more reliable.  The tfm in the projects, is probably still usefull, for when the developer is messing with that code.

I believe the -f flows to leaf projects, at least when I do it on the command line I get this

dotnet build src\buildtools\buildtools.proj -restore -f netcoreapp3.1

````
dotnet build src\buildtools\buildtools.proj -restore -f netcoreapp3.1
Microsoft (R) Build Engine version 16.5.0+d4cbfca49 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

   **.... stuff elided**

  FSharp.Core -> c:\kevinransom\fsharp\artifacts\bin\FSharp.Core\Debug\netstandard2.0\FSharp.Core.dll
  fslex -> c:\kevinransom\fsharp\artifacts\bin\fslex\Debug\**netcoreapp3.1**\fslex.dll
  fsyacc -> c:\kevinransom\fsharp\artifacts\bin\fsyacc\Debug\**netcoreapp3.1**\fsyacc.dll
  AssemblyCheck -> c:\kevinransom\fsharp\artifacts\bin\AssemblyCheck\Debug\**netcoreapp3.1**\AssemblyCheck.dll

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:11.17

````